lib/JSON/Plugin.php

```diff
@@ -800,6 +800,35 @@ function isHiddenPrivateEvent($vevent, $node) {
         return $vevent->CLASS == 'PRIVATE' && $node->getOwner() != $this->currentUser;
     }
 
+    function handleJsonRequest($path, $node, $jsonData) {
+        if (isset($jsonData->{'invite-reply'})) {
+            return $this->updateInviteStatus($path, $node, $jsonData);
+        } else if (isset($jsonData->share)) {
+            return $this->updateSharees($path, $node, $jsonData);
+        }
+    }
+
+    function updateInviteStatus($path, $node, $jsonData) {
+        if (isset($jsonData->{'invite-reply'})) {
+            if(isset($jsonData->{'invite-reply'}->invitestatus)) {
+                switch ($jsonData->{'invite-reply'}->{'invitestatus'}) {
+                    case 'accepted': 
+                        $inviteStatus = \Sabre\DAV\Sharing\Plugin::INVITE_ACCEPTED;
+                        break;
+                    case 'noresponse':
+                        $inviteStatus = \ESN\DAV\Sharing\Plugin::INVITE_NORESPONSE;
+                }
+            }
+
+            $node->updateInviteStatus($inviteStatus);
```

Comment:
So if `isset($jsonData->{'invite-reply'}->invitestatus) === false` we still call $node->updateInviteStatus($inviteStatus)  with `$invitestatus` variable not defined as you define it in the if block above

Should be with the first conditional : `isset($jsonData->{'invite-reply'}) && $jsonData->{'invite-reply'}->{'invitestatus'}`